### PR TITLE
[RFR] Reorganize webpack configurations

### PR DESCRIPTION
**Pull request by djhi**
- [x] Extracted webpack config for each project in the project directory
- [x] Separated webpack procces for frontend and admin thus simplifying build process (path do not change depending on environment anymore)

Having separate webpack also means apps are served on different ports. One effect is that being logged in admin does not mean being logged on frontend anymore

### CODE REVIEW

**Comment by fzaninotto** [comment]
needs rebase, too

**Comment by djhi** [comment]
@Kmaschta no the build structure has not changed so its just a matter of configuration for prod and staging. Right ?

**Comment by Kmaschta** [comment]
I just know that if you want to deploy the admin in the `admin` folder of a bucket, you need change the publicPath to `/admin`.

File: config/pm2_servers/dev.json [hunk 1]
```diff
@@ -31,6 +31,23 @@
             "instances": 1,
             "name": "bpm_frontend-dev",
             "script": "./node_modules/.bin/webpack-dev-server"
-        }
-    ]
+        }, {
+            "args": [
+                "--config=./src/admin/webpack.config.babel.js",
+                "--content-base=build/admin",
+                "--devtool=cheap-module-inline-source-map",
```
Comment (brikou) [inline config/pm2_servers/dev.json]: We have here webpack configuration, this should be done (as much as possible) in a single place (webpack.config.js) to make it easy to read.

File: src/admin/webpack.config.babel.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    entry: {
+        index: [
+            resolve(__dirname, './js/main.js'),
+        ],
+    },
+    module: {
+        loaders: [{
+            test: /\.jsx?$/,
+            exclude: /node_modules/,
+            loader: 'babel-loader',
```
Comment (brikou) [inline src/admin/webpack.config.babel.js]: you can just write `babel`

File: src/admin/webpack.config.babel.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+            // Options to configure babel with
+            query: {
+                cacheDirectory: true,
+                plugins: [
+                    'transform-react-jsx',
+                    ['transform-runtime', {
+                        polyfill: false,
+                        regenerator: true,
+                    }],
+                    'add-module-exports',
```
Comment (brikou) [inline src/admin/webpack.config.babel.js]: Using this plugin is required when switching from babel 5 to 6 (and we don't want to change codebase) but we should get rid of it (and append `.default` when needed). Less plugins === less maintenance too.

File: src/admin/webpack.config.babel.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    module: {
+        loaders: [{
+            test: /\.jsx?$/,
+            exclude: /node_modules/,
+            loader: 'babel-loader',
+            // Options to configure babel with
+            query: {
+                cacheDirectory: true,
+                plugins: [
+                    'transform-react-jsx',
```
Comment (brikou) [inline src/admin/webpack.config.babel.js]: We already have react preset, so I don't understand this requirement.

File: src/admin/webpack.config.babel.js [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+                    ['transform-runtime', {
+                        polyfill: false,
+                        regenerator: true,
+                    }],
+                    'add-module-exports',
+                ],
+                presets: [
+                    'es2015',
+                    'react',
+                    'stage-0',
```
Comment (brikou) [inline src/admin/webpack.config.babel.js]: I think stage-1 is enough (leaving on the edge but not too much :) )

File: src/admin/webpack.config.babel.js (hunk 3 above)
Comment (djhi) [inline src/admin/webpack.config.babel.js]: Yes but I really don't like this `.default` thing. I'll wait for more feedback from the others

File: src/admin/webpack.config.babel.js (hunk 5 above)
Comment (djhi) [inline src/admin/webpack.config.babel.js]: agreed, moreover, stage-0 doesn't bring much anyway

File: src/admin/webpack.config.babel.js (hunk 4 above)
Comment (djhi) [inline src/admin/webpack.config.babel.js]: learned something :)

File: src/admin/webpack.config.babel.js [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+export default {
+    entry: {
+        index: [
+            resolve(__dirname, './js/main.js'),
+        ],
+    },
+    module: {
+        loaders: [{
+            test: /\.jsx?$/,
+            exclude: /node_modules/,
```
Comment (djhi) [inline src/admin/webpack.config.babel.js]: :+1:

File: config/pm2_servers/dev.json (hunk 1 above)
Comment (djhi) [inline config/pm2_servers/dev.json]: I haven't been able to move everything in the webpack config. `hot` and `port` don't work

File: src/admin/webpack.config.babel.js (hunk 3 above)
Comment (Kmaschta) [inline src/admin/webpack.config.babel.js]: I'm in favor of keep this config, but what about move it in the `.babelrc` file?

Comment (djhi) [inline src/admin/webpack.config.babel.js]: The babelrc file is in fact only used by eslint plugin. It is completely overridden for webpack.

Comment (Kmaschta) [inline src/admin/webpack.config.babel.js]: Ok ! Not the point of the PR

**Pull request merged**